ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -504,6 +504,47 @@ private CompactionType determineCompactionType(CompactionInfo ci, AcidDirectory
       if (initiateMajor) return CompactionType.MAJOR;
     }
 
+    // bucket size calculation can be resource intensive if there are numerous deltas, so we check for rebalance
+    // compaction only if the table is in an acceptable shape: no major compaction required. This means the number of
+    // files shouldn't be too high
+    if ("tez".equalsIgnoreCase(HiveConf.getVar(conf, HiveConf.ConfVars.HIVE_EXECUTION_ENGINE)) &&
```

Comment:
Major->Rebalance->Minor. As I see, after the change, that is the priority order of the different type of compactions. Is that the right order? Does the order matter at all?

Reply:
Yes, running a rebalance compaction on uncompacted tables could be resource intensive due to the hive number of files and folders. So I decided to schedule rebalance compactions only on tables already major compacted. This ensures that the number of deltas are relatively low.